components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/domain/InMemoryDomainManager.java

```diff
@@ -18,95 +18,226 @@
-            String identityStoreConnectorId, String domainName) throws DomainManagerException {
+            String identityStoreConnectorId, String domainName) throws DomainException {
 
-        Domain domain = getDomainFromName(domainName);
+        Domain domain = this.getDomainFromDomainName(domainName);
 
         IdentityStoreConnector identityStoreConnector = domain.getIdentityStoreConnectorFromId
                 (identityStoreConnectorId);
 
-        if (null != identityStoreConnector) {
-            return identityStoreConnector;
-        } else {
-            throw new DomainManagerException("IdentityStoreConnector " + identityStoreConnectorId +
-                    " was not found ");
+        if (identityStoreConnector == null) {
+            throw new DomainException(String
+                    .format("IdentityStoreConnector %s was not found", identityStoreConnectorId));
         }
 
+        return identityStoreConnector;
     }
 
     @Override
     public Map<String, IdentityStoreConnector> getIdentityStoreConnectorMapForDomain(
-            String domainName) throws DomainManagerException {
+            String domainName) throws DomainException {
 
-        return getDomainFromName(domainName)
+        return this.getDomainFromDomainName(domainName)
                 .getIdentityStoreConnectorMap();
 
     }
 
+    @Override
+    public void addCredentialStoreConnectorToDomain(
+            CredentialStoreConnector credentialStoreConnector,
+            String domainName) throws DomainException {
+
+        Domain domain = this.getDomainFromDomainName(domainName);
+
+        domain.addCredentialStoreConnector(credentialStoreConnector);
+    }
+
+    @Override
+    public CredentialStoreConnector getCredentialStoreConnector(
+            String credentialStoreConnectorId, String domainName) throws DomainException {
+
+        Domain domain = this.getDomainFromDomainName(domainName);
+
+        CredentialStoreConnector credentialStoreConnector = domain
+                .getCredentialStoreConnectorFromId(credentialStoreConnectorId);
+
+        if (credentialStoreConnector == null) {
+            throw new DomainException(String
+                    .format("credentialStoreConnector %s was not found", credentialStoreConnectorId));
+        }
+
+        return credentialStoreConnector;
+    }
+
+    @Override
+    public Map<String, CredentialStoreConnector> getCredentialStoreConnectorMapForDomain(
+            String domainName) throws DomainException {
+
+        return this.getDomainFromDomainName(domainName)
+                .getCredentialStoreConnectorMap();
+    }
+
+    @Override
+    public Domain getDomainFromClaim(Claim claim) throws DomainException {
+
+        Set<Map.Entry<Integer, Map<String, Domain>>> set = domainPriorityToDomainMap.entrySet();
+
+        for (Map.Entry<Integer, Map<String, Domain>> entry : set) {
+
+            for (Map.Entry<String, Domain> domainMapEntry : entry.getValue().entrySet()) {
```

Comment:
Since key of this entry is not used we can iterate through entry values. Also check if we can avoid having a map inside a map.